pkg/daemon/ceph/client/pool.go

```diff
@@ -137,10 +137,16 @@ func CreatePoolWithProfile(context *clusterd.Context, clusterName string, newPoo
 	isReplicatedPool := newPool.ErasureCodeProfile == "" && newPool.Size > 0
 	if isReplicatedPool {
 		return CreateReplicatedPoolForApp(context, clusterName, newPool, appName)
-	} else {
-		// If the pool is not a replicated pool, then the only other option is an erasure coded pool.
-		return CreateECPoolForApp(context, clusterName, newPool, appName, true /* enableECOverwrite */)
 	}
+	// If the pool is not a replicated pool, then the only other option is an erasure coded pool.
+	return CreateECPoolForApp(
+		context,
+		clusterName,
+		newPool,
+		appName,
+		true, /* enableECOverwrite */
+		uint64(newPoolReq.ErasureCodedConfig.DataChunkCount),
```

Comment:
Minor suggestion: pass the whole `newPoolReq.ErasureCodedConfig` object so that if the function needs more config options for another reason in the future, it will already be available. 

Alternatively, I would suggest leaving it as a uint parameter, then cast later when it is needed by `FormatUint`. I think it would be a small help with code readability.